projects/opentrain/gtfs_pipeline/heatmap/HeatMap.java

```diff
@@ -0,0 +1,895 @@
+          float value = (float) (heatMap[(int) (longitude * 100) - 1][(int) (latitude * 100) - 1] - minData) / (float) (maxData - minData);
+          int r =
+                  (int)
+                          (((float) color2.getRed() - (float) color1.getRed()) * value
+                                  + (float) color1.getRed());
+          int g =
+                  (int)
+                          (((float) color2.getGreen() - (float) color1.getGreen()) * value
+                                  + (float) color1.getGreen());
+          int b =
+                  (int)
+                          (((float) color2.getBlue() - (float) color1.getBlue()) * value
+                                  + (float) color1.getBlue());
+          if ((r > 0 && r < 255) && (g > 0 && g < 255) && (b > 0 && b < 255)) {
+            System.out.println("r=" + r + "b=" + b + "g" + g);
+            Color color = new Color(r, g, b, 50);
+            graphics2D.setPaint(color);
+            graphics2D.fillRect(i, j, 5, 5);
+          }
+        }
+      }
+    }
+    //X on stops
+      for (StopsAround stp : stopsAroundList) {
+        if (stp.i == i && stp.j == j) {
+          x =
+          getXpixel(stp.stop.getStopLat(),stp.stop.getStopLon());
+          y = getYpixel(stp.stop.getStopLat(),stp.stop.getStopLon());
+          graphics2D.setPaint(Color.black);
+          graphics2D.setFont(new Font("TimesRoman",Font.BOLD,28));
+          graphics2D.drawString("X",x, y);
+          // }
+          break;
+        }
+      }
+
+    graphics2D.dispose();
+    // Save file
+    boolean writeOnImage = false;
+    try {
+      writeOnImage =
+              ImageIO.write(
+                      image,
+                      "png",
+                      new File(
+                              heatMapPath.get()));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    if (!writeOnImage) {
+      System.out.println("can't write on image");
+    }
+    System.out.println("saved");
+  }*/
+
+  public static void main(String[] args) throws IOException {
+    // TODO:
+    //      Heatmap matrix size to max mapsize
+    //      Boarders of israel,do not paint sea area
+    //      Improve x,y pixel method by using more data
+    //      Change i,j in StopsAround to x,y
+    //      Fix latitude longitude names
+
+    /** To use change output path on drawOnRealMap */
+    Flags.parseCurrentPackage(args);
+    StopTime stopTime;
+    Trip trip;
+
+    try (FileInputStream tripInputStream =
+            new FileInputStream(new File("projects/opentrain/gtfs_pipeline/trips.protobin"));
+        FileInputStream stopTimeInputStream =
+            new FileInputStream(new File("projects/opentrain/gtfs_pipeline/stopTime.protobin"))) {
+
+      // Test some trip stops heatMap
+      trip = Trip.parseDelimitedFrom(tripInputStream);
+      stopTime = StopTime.parseDelimitedFrom(stopTimeInputStream);
+
+      while (stopTime != null | trip != null) {
```

Comment:
Why do you need to check `trip != null` every time if `trip` is not changed in the loop?

Reply:
If the trip file is empty i don't start the loop,
i added if for that case in the new HeatMap.java.